### ISSUE TITLE
[rollup-plugin] Incorrect file name prevents uploading of sourcemap

### ISSUE THREAD

**Issue by joekrump** [honeybadger-io/honeybadger-js#1062]
## What are the steps to reproduce this issue?
1. I am using Vite in a Ruby on Rails app that uses the `vite_rails` gem
2. I have some `.vue` code that is in a `app/frontend` directory
3. I have some entrypoint files for vite which are in `app/frontend/entrypoints`:
     1.  `app/frontend/entrypoints/application.js`
     2. `app/frontend/entrypoints/quill.js` 
4. My build runs on a Heroku Ruby buildpack
5. The command that runs the build is `rails assets:precompile` which calls `bin/vite build`

## What happens?

```bash
[honeybadger] Failed to upload 1 sourcemap file(s) to Honeybadger
remote:        Error: ENOENT: no such file or directory, open '/tmp/build_9f87aa0c/public/vite/assets/assets/application-fa0ba0ae.js'
remote:        error during build:
remote:        Error: Failed to upload 1 sourcemap file(s) to Honeybadger
remote:        Error: ENOENT: no such file or directory, open '/tmp/build_9f87aa0c/public/vite/assets/assets/application-fa0ba0ae.js'
```

Cannot find the application.js source file because the path it is looking at does not exist (note the `assets/assets` bit).

The other sourecmaps files are successfully uploaded, just not this one. 😕 

## What were you expecting to happen?

All sourcemaps would be uploaded.

## Any logs, error output, etc?
If I add some logging to https://github.com/honeybadger-io/honeybadger-js/blob/406488423cfd4995967d7044f620c753d6896a13/packages/rollup-plugin/src/rollupUtils.ts#L49-L52

```javascript
console.warn("path.dirname(sourcemapFilename): ", path.dirname(sourcemapFilename))
console.warn("file: ", file)
console.warn("jsFilename: ", jsFilename)
console.warn("**********************************");
```

I can see that the `file` value for my application.js file begins with `assets/` whereas all the others do not. 

Logged output (truncated):

```
path.dirname(sourcemapFilename):  assets
file:  quill-eee79fc4.js
jsFilename:  assets/quill-eee79fc4.js
**********************************
path.dirname(sourcemapFilename):  assets
file:  AwardForm-ce117868.js
jsFilename:  assets/AwardForm-ce117868.js
**********************************
path.dirname(sourcemapFilename):  assets
file:  ProjectCategoryFormPanel-959db9fd.js
jsFilename:  assets/ProjectCategoryFormPanel-959db9fd.js
**********************************
path.dirname(sourcemapFilename):  assets
file:  HackathonTemplateSelect-dcf2d319.js
jsFilename:  assets/HackathonTemplateSelect-dcf2d319.js
**********************************
path.dirname(sourcemapFilename):  assets
file:  TimePicker-7a758387.js
jsFilename:  assets/TimePicker-7a758387.js
**********************************
path.dirname(sourcemapFilename):  assets
file:  assets/application-6016d30a.js
jsFilename:  assets/assets/application-6016d30a.js
**********************************
```

## Any other comments?

The name of the extra nested directory is the value of `build.assetsDir` set in my vite config.

<details>
  <summary><strong>ℹ️ vite.config.ts</strong></summary>
  
   ```typescript
      import { defineConfig } from 'vite';
      import rubyPlugin from 'vite-plugin-ruby';
      import vue from '@vitejs/plugin-vue';
      import svgLoader from 'vite-svg-loader';
      import fullReload from 'vite-plugin-full-reload';
      import honeybadgerRollupPlugin from '@honeybadger-io/rollup-plugin';
      
      const isStaging = process.env.RAILS_ENV === 'staging';
      const isProduction = process.env.RAILS_ENV === 'production';
      
      const HONEYBADGER_API_KEY = JSON.stringify(process.env.HONEYBADGER_FRONTEND_API_KEY)?.trim();
      const HONEYBADGER_REVISION = JSON.stringify(process.env.SOURCE_VERSION)?.trim();
      const ENVIRONMENT = JSON.stringify(process.env.ENVIRONMENT)?.trim();
      // eslint-disable-next-line no-relative-import-paths/no-relative-import-paths, import/extensions
      import { RESOLUTION_ALIASES } from './config/vite';
      
      const vitePlugins = [
        // Do a full reload when changes are made to files with paths that match these.
        fullReload(['app/views/**/*']),
        rubyPlugin(),
        svgLoader(),
        vue(),
      ];
      const rollupPlugins = [];
      
      if (isProduction || isStaging) {
        const honeybadgerPluginOptions = {
          apiKey: process.env.HONEYBADGER_FRONTEND_API_KEY ?? "hbp_testtesttest",
          assetsUrl: isStaging ? 'https://assets.stating.com' : 'https://assets.production.com',
          revision: process.env.SOURCE_VERSION,
        };
      
        // used for uploading sourcemaps to honeybadger to allow for better debugging
        rollupPlugins.push(honeybadgerRollupPlugin(honeybadgerPluginOptions));
      }
      
      export default defineConfig({
        build: {
          sourcemap: isStaging || isProduction ? 'hidden' : 'inline',
          rollupOptions: {
            plugins: rollupPlugins,
          },
        },
        resolve: {
          alias: RESOLUTION_ALIASES,
        },
        define: {
          HONEYBADGER_API_KEY,
          ENVIRONMENT,
          HONEYBADGER_REVISION,
        },
        plugins: vitePlugins,
      });
   ```
</details>

I can fix the issue for myself if I update `formatSourcemapData` to this: https://github.com/honeybadger-io/honeybadger-js/commit/4ba311942a3808b73807350f6a74250315959817

_I would still like to figure out what the root cause is though, and it is possible it could be a bug in sourcemap generation code in rollup itself._

## What versions are you using?
**Operating System:** Mac OSX
**Package Name:** `@honeybadger-io/rollup-plugin`
**Package Version:** 5.4.0
**Browser Version:** N/A

**Comment by joekrump** [comment]
If I set `build.assetsDir` in my vite config file to `''` then the `file` value in the sourcemap for `application.js` is correct. Weird that this is only for this one file though.

**Comment by BethanyBerkowitz** [comment]
Thanks, @joekrump, this is really interesting. When setting this up originally, I found experimentally that the value for `file` did not include folder path, but clearly that's not a reliable assumption. I'm in the same boat as you right now -- I don't understand why one file would be treated differently than the rest in the bundle. 

I can put out a small update to just check if the subfolder is in the `file` value already before adding it in, but I agree with you that I'd rather try to learn the root cause rather than just add a workaround. So I'm going to set up a skeleton of a `vite_rails` project to play with, unless you already have one I can look at?

**Comment by joekrump** [comment]
>So I'm going to set up a skeleton of a vite_rails project to play with, unless you already have one I can look at?

I don't have one at the moment but am happy to help here so let me know if you get stuck or cannot repro. In my private repo I'm using the latest versions of vite, rollup, and the vite_rails gem (which includes vite_ruby) and have followed all the default setup for frontend assets so the entrypoints are in `app/frontend/entrypoints`

**Comment by BethanyBerkowitz** [comment]
Hey @joekrump, I created a basic vite_rails app by first following the [Rails setup guide](https://guides.rubyonrails.org/getting_started.html) and then following the [Vite Ruby instructions](https://vite-ruby.netlify.app/guide/). I added a second entrypoint file (`foo.js`) and both that file and the `application.js` file seemed to work as expected. I didn't get the duplicate `assets/` folder issue. 

It's maybe worth noting that in following the default instructions, my files ended up in `app/javascript/entrypoints`. However, I was able to change this to `app/frontend/entrypoints` by renaming the folder and setting the `sourceCodeDir` in an environment variable, eg `VITE_RUBY_SOURCE_CODE_DIR=app/frontend rails assets:precompile`. 

<img width="680" alt="Screen Shot 2023-05-03 at 3 41 42 PM" src="https://user-images.githubusercontent.com/16493493/236030180-40f7f9b8-ba82-48fc-b477-6e59a4054067.png">

I think my setup is pretty close to yours, but definitely not exactly the same. A few questions: 

1. If you run the build locally, do you still have this issue? Is it specific to running in a Heroku buildpack? 
2. In your `vite.config.ts`, what is the purpose of your `RESOLUTION_ALIASES`? It doesn't seem obviously related, but it's also something I can't replicate since I don't know what you're using that for. 

Thanks!

**Comment by joekrump** [comment]
>If you run the build locally, do you still have this issue? Is it specific to running in a Heroku buildpack?

I do hit it locally as well.

>In your vite.config.ts, what is the purpose of your RESOLUTION_ALIASES? It doesn't seem obviously related, but it's also something I can't replicate since I don't know what you're using that for.

Oh ya, here's what I've got in the `vite.ts` file that I get `RESOLUTION_ALIASES` from:
```typescript
import path from 'path';

const RESOLUTION_ALIASES = {
  '@': path.resolve(__dirname, '../app/frontend/src'),
  '@bold-icons': path.resolve(__dirname, '../app/frontend/images/streamlinehq_icons/bold'),
  '@devpost': '@devpost',
  '@fonts': path.resolve(__dirname, '../app/frontend/fonts'),
  '@images': path.resolve(__dirname, '../app/frontend/images'),
  '@light-icons': path.resolve(__dirname, '../app/frontend/images/streamlinehq_icons/light'),
  '@store_modules': path.resolve(__dirname, '../app/frontend/src/stores/store_modules'),
  '@stylesheets': path.resolve(__dirname, '../app/frontend/stylesheets'),
};

export { RESOLUTION_ALIASES };
```

**Comment by subzero10** [comment]
@joekrump This should be fixed with `v5.5.0`!

Thanks @BethanyBerkowitz!
